modules/votes/hooks/useVoteActionsModals.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useEffect, useMemo } from 'react'
+import { useVoteActionModal } from 'modules/votes/hooks/useVoteActionModal'
+import { VoteSubmitModal } from 'modules/votes/ui/VoteActionsModals/SubmitModal/VoteSubmitModal'
+import { VoteSuccessModal } from 'modules/votes/ui/VoteActionsModals/SuccessModal/VoteSuccessModal'
+import { useVoteFormActionsContext } from 'modules/votes/providers/VoteFormActions/VoteFormActionsContext'
+
+import { VoteMode } from 'modules/votes/types'
+import { ResultTx } from 'modules/blockChain/types'
+
+type ModalKey = 'submit' | 'success'
+
+type ModalData = {
+  mode?: VoteMode
+  successTx?: ResultTx
+}
+
+export function useVoteActionsModals() {
```

Comment:
It seems a bit excessive for me to have another modal-managing system like this.
I would like to suggest considering to stick with the system already existing in the project, doing something like that:

```js
import { getUseModal } from 'modules/modal/useModal'
import { VoteSubmitModal } from 'modules/votes/ui/VoteActionsModals/SubmitModal/VoteSubmitModal'
import { VoteSuccessModal } from 'modules/votes/ui/VoteActionsModals/SuccessModal/VoteSuccessModal'

export const useSubmitModal = getUseModal(VoteSubmitModal)
export const useSuccessModal = getUseModal(VoteSuccessModal)
```

If it lacks of some necessary features, like not supporting modal switching or something like that, I suggest to check out our more latest and advanced implementation of `ModalProvider` here: https://github.com/lidofinance/ethereum-staking-widget/blob/develop/providers/modal-provider.tsx

My idea is about upgrading existing system with lacking features over than extending the codebase with a similar system.

Reply:
fixed in f675775e809797ec0e4140388b920cf937642478